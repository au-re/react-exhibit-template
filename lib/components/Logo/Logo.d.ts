/// <reference types="react" />
import "./Logo.css";
/** The Exhibit logo with some animations. Demoing usage with css files */
interface ILogo {
    size: string;
}
declare const Logo: (props: ILogo) => JSX.Element;
export default Logo;
