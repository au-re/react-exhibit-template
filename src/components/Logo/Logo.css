.Exhibit__Logo__Box {
  width: 0px;
  height: 0px;
  cursor: pointer;
}

.Exhibit__Logo__Box.Box--Outer {
  box-shadow: 2px 2px 4px rgba(0, 0, 0, .2);
  position: relative;
  border-bottom: 120px solid #29ACE2;
  border-left: 120px solid #7454A2;
}

.Exhibit__Logo__Box.Box--Inner {
  box-shadow: 2px 2px 4px rgba(0, 0, 0, .2);
  position: absolute;
  margin-left: -100px;
  margin-top: 20px;
  border-bottom: 80px solid #F8932B;
  border-left: 80px solid #F9AE15;
  transition: transform .25s ease-in-out;
}

.Exhibit__Logo__Box.Box--Inner:hover {
  transform: rotate(90deg);
}

.Exhibit__Logo__Box.Box--Inner:hover+.Exhibit__Logo__Circle.Circle--Top {
  transform: translate(0px, 20px);
}

.Exhibit__Logo__Circle {
  box-shadow: 2px 2px 4px rgba(0, 0, 0, .2);
  position: absolute;
  z-index: 4000;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.Exhibit__Logo__Circle.Circle--Top {
  margin-top: 92px;
  margin-left: -68px;
  background-color: #EB529A;
  transition: transform .25s ease-in-out;
}