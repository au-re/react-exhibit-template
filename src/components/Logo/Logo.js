import "./Logo.css";

import React from "react";

/** The Exhibit logo with some animations. Demoing usage with css files */

const Logo = () =>
  (<div className="Exhibit__Logo__Box Box--Outer">
    <div className="Exhibit__Logo__Box Box--Inner" />
    <div className="Exhibit__Logo__Circle Circle--Top" />
  </div>);

export default Logo;
