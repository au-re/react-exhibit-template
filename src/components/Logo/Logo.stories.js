import { storiesOf } from "@storybook/react";
import React from "react";
import { withDocs } from "storybook-readme";

import Logo from "./Logo";

import readme from "./README.md";

storiesOf("Logo", module)
  .addDecorator(withDocs(readme))
  .add("project logo", () => <Logo />);
