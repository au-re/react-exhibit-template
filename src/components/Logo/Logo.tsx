import "./Logo.css";

import React from "react";

/** The Exhibit logo with some animations. Demoing usage with css files */

interface ILogo {
  size: string;
}

const Logo = (props: ILogo) =>
  (<div className="Exhibit__Logo__Box Box--Outer">
    <div className="Exhibit__Logo__Box Box--Inner" />
    <div className="Exhibit__Logo__Circle Circle--Top" />
  </div>);

export default Logo;
