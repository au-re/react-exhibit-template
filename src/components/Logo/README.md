Logo
===

The Exhibit logo with some animations. Demoing usage with css files.

```jsx
  <Logo />
```

<!-- STORY -->