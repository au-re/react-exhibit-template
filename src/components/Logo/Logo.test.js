import "jest-styled-components";

import React from "react";
import renderer from "react-test-renderer";

import Logo from "./Logo";

describe("Logo", () => {

  it("renders without props", () => {
    const tree = renderer.create(<Logo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
