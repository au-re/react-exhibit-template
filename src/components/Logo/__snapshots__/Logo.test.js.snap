// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> renders without props 1`] = `
<div
  className="Exhibit__Logo__Box Box--Outer"
>
  <div
    className="Exhibit__Logo__Box Box--Inner"
  />
  <div
    className="Exhibit__Logo__Circle Circle--Top"
  />
</div>
`;
